test/unit/many_to_many_test.rb

```diff
@@ -2,13 +2,17 @@
 require 'active-record-ex/many_to_many'
 
 class ManyToManyTest < ActiveSupport::TestCase
+  # Declare classes first to avoid circular dependencies
+  class HasManied < StubModel; end
+  class SomeClassNamedByClass < StubModel; end
   class HasManied < StubModel
     include ActiveRecordEx::ManyToMany
 
     has_one :one
     has_many :simple_belongs_tos
     has_many :belongs_to_throughs, through: :simple_belongs_tos
     has_many :class_nameds, class_name: 'ManyToManyTest::SomeClassName'
+    has_many :class_nameds_by_class, class_name: ManyToManyTest::SomeClassNamedByClass
```

Comment:
I think this needs to be changed to `class_named_by_classs`? Pretend you're a snake!